### PR TITLE
IR-3-implementing-updating-an-incident

### PR DESCRIPTION
**Pull request by ttcollins**
Pull this branch and run the `IReporter.java` file

### CODE REVIEW

**Comment by ttcollins** [comment]
Sorry i had to delete the old branch, it caused issues with my IDE

File: src/org/pahappa/systems/services/IncidentServiceImpl.java [hunk 1]
```diff
@@ -41,7 +41,14 @@ public Incident saveIncident(Incident incident) throws Exception {
 
 	@Override
 	public Incident updateIncident(Incident incident) throws Exception {
-		// TODO Auto-generated method stub
+		for(Incident item: incidents){
+			if(item.getId() == incident.getId()){
+				item.setTitle(incident.getTitle());
+				item.setType(incident.getType());
+				item.setComment(incident.getComment());
+				return item;
```
Comment (katusiimeconrad) [inline src/org/pahappa/systems/services/IncidentServiceImpl.java:49]: I think If the incident was found, you're supposed to throw an Exception, that the Update failed.

**Review by katusiimeconrad — COMMENTED** [review]
All is good, The update works, except only for the Exception part.

File: src/org/pahappa/systems/services/IncidentServiceImpl.java (hunk 1 above)
Comment (ttcollins) [reply src/org/pahappa/systems/services/IncidentServiceImpl.java:49]: Totally skipped my mind, thanx

**Comment by ttcollins** [comment]
> All is good, The update works, except only for the Exception part.

I have fixed this

**Review by katusiimeconrad — APPROVED** [review]
Great job Sir!
All is good

**Review by iandrew8 — APPROVED** [review]

**Review by NakagweSharifah — APPROVED** [review]

**Review by Golder12 — APPROVED** [review]

**Review by reifred — APPROVED** [review]

**Pull request merged**